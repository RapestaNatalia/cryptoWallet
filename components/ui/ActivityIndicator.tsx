import { ActivityIndicator } from "react-native";
import React from "react";
interface Fetching {
  animating: boolean;
  color: string;
}
export const Fetching = ({ animating, color }: Fetching) => {
  return <ActivityIndicator size="large" animating={animating} color={color} />;
};
