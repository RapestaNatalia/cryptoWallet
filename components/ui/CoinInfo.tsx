import React from "react";
import { Item } from "../../interfaces/interfaces";
import { StyleSheet, Text, View, TouchableWithoutFeedback } from "react-native";
import ArrowComponent from "../icons/ArrowIcon";
import { formatUSA } from "../../config/formatNumber";
import { theme } from "../../style/theme";
export interface Coin {
  item: Item;
  onPress?: () => void;
  styleItem?: Object;
}
export const CoinInfo = ({
  item,
  onPress,
  styleItem,
}:Coin) => {
  const priceUsd = parseFloat(item.priceUsd).toFixed(2);
  const changePercent24Hr = (+item.changePercent24Hr).toFixed(1);
  const isPositive = +changePercent24Hr >= 0 ? true : false;

  return (
    <View style={styles.mainContainer}>
      <View style={styles.itemContainer}>
        <TouchableWithoutFeedback onPress={onPress}>
          <View style={styleItem}>
            <View style={styles.header}>
              <View style={styles.leftContainer}>
                <Text style={styles.money}>{item.symbol} - </Text>
                <Text style={styles.money}>{item.name}</Text>
              </View>
              <View style={styles.rigthContainerTop}>
                <Text style={styles.rank}>#{item.rank}</Text>
              </View>
            </View>
            <View style={styles.bottomDescription}>
              {/* ðŸ’¯  In this execercise you can round numbers without a library */}
              <View style={styles.leftContainer}>
                <Text style={styles.price}>
                  {/* TODO agregar formato */}$ {formatUSA(priceUsd)}
                  <Text style={styles.usd}> USD</Text>
                </Text>
              </View>
              <View
                style={[
                  styles.rigthContainer,
                  styles.percentageContainer,
                  isPositive ? styles.boxPositive : styles.boxNegative,
                  styles.box,
                ]}
              >
                <View style={!isPositive && styles.arrowNegative}>
                  <ArrowComponent
                    color={
                      isPositive
                        ? theme.colors.greenCrayola
                        : theme.colors.deepCarminePink
                    }
                  />
                </View>
                <Text style={styles.rigthContent}>
                  {" "}
                  {Math.abs(+changePercent24Hr)}%
                </Text>
              </View>
            </View>
          </View>
        </TouchableWithoutFeedback>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  mainContainer: {
    backgroundColor: "white",
    borderRadius:8,
  },
  header: {
    flexDirection: "row",
    flex: 1,
  },
  illustration: {
    width: 50,
    height: 50,
  },
  itemContainer: {
    borderRadius: 8,
    display: "flex",
    backgroundColor: theme.colors.white,
    marginVertical: 6,
    padding: 8,
  },
  bottomDescription: {
    flexDirection: "row",
    flex: 1,
  },
  rigthContent: {
    justifyContent: "flex-end",
    alignItems: "flex-end",
  },
  leftContent: {
    alignItems: "flex-start",
  },
  price: {
    color: theme.colors.bondiBlue,
    fontSize: 24,
    fontWeight: "normal",
  },
  usd: {
    color: theme.colors.auroMetalSaurus,
    fontSize: 14,
    textAlignVertical: "bottom",
  },
  money: {
    color: theme.colors.zinnwalditeBrown,
    fontSize: 16,
    fontWeight: "bold",
    fontStyle: "normal",
  },
  rank: {
    color: theme.colors.auroMetalSaurus,
    fontSize: 14,
    fontStyle: "normal",
    alignSelf: "flex-end",
  },
  leftContainer: {
    flexDirection: "row",
    flex: 1,
    justifyContent: "flex-start",
  },
  rigthContainerTop: {},
  rigthContainer: {
    flexDirection: "row",
    justifyContent: "flex-end",
  },
  percentageContainer: {
    alignItems: "flex-end",
  },
  box: {
    paddingVertical: 2,
    paddingHorizontal: 10,
    borderRadius: 12,
    fontSize: 14,
    fontStyle: "normal",
    fontWeight: "500",
    alignItems: "center",
  },
  boxPositive: {
    backgroundColor: theme.colors.aeroBlue,
    color: theme.colors.castletonGreen,
  },
  boxNegative: {
    backgroundColor: theme.colors.palePink,
    color: theme.colors.darkCandyAppleRed,
  },
  arrowNegative: {
    transform: [{ rotate: "180deg" }],
  },
});
