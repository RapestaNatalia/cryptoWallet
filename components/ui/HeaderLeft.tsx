import React from "react";
import { StyleSheet, TouchableOpacity, Text, View } from "react-native";
import ChevronLeft from "../icons/ChevronLeft";
import { theme } from "../../style/theme";

export interface Header {
  onPress: () => void;
  name: string;
}
export const HeaderLeft = ({ onPress, name }: Header) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={styles.container}>
        <ChevronLeft />
        <Text style={styles.text}>{name}</Text>
      </View>
    </TouchableOpacity>
  );
};
const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    alignItems: "center",
    paddingLeft: 10,
  },
  text: {
    color: theme.colors.bondiBlue,
    fontSize: 14,
    marginLeft: 7,
  },
});
