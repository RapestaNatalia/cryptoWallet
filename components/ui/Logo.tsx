import * as React from "react";
import { Image } from "react-native";

export const Logo = () => {
  return (
    <Image
      style={{ width: 28.53, height: 26.07 }}
      source={require("../../assets/logo.png")}
    />
  );
};
