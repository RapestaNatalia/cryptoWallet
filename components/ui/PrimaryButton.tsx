import React from "react";
import { StyleSheet, TouchableOpacity, Text, View } from "react-native";
import { theme } from "../../style/theme";

export interface PrimaryBtn {
  title: string;
  onPress: () => void;
  accessibilityLabel?: string;
  accessibilityHint?: string;
}
export const PrimaryButton = ({
  title,
  onPress,
  accessibilityHint,
  accessibilityLabel,
}: PrimaryBtn) => {
  return (
    <View
      style={{
        backgroundColor: theme.colors.batteryChargedBlue,
        height: 42,
        borderRadius: 6,
        width: "100%",
        marginTop: 44,
      }}
    >
      <TouchableOpacity
        onPress={onPress}
        accessibilityLabel={accessibilityLabel}
        accessibilityHint={accessibilityHint}
      >
        <Text style={styles.textStyle}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};
const styles = StyleSheet.create({
  button: {
    color: theme.colors.batteryChargedBlue,
  },
  textStyle: {
    color: theme.colors.white,
    fontSize: 16,
    textAlign: "center",
    padding: 9,
  },
});
