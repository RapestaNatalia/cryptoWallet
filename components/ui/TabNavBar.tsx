import { StyleSheet, View, Text } from "react-native";
import React from "react";
import { theme } from "../../style/theme";

export interface props {
  name: string;
  isFocus: boolean;
}
export const TabNavBar = ({ name, isFocus }: props) => {
  return (
    <View style={[styles.container, { borderTopWidth: isFocus ? 3 : 0 }]}>
      <Text style={[styles.labelStyle,isFocus?styles.activeColor:styles.inactiveColor]}>{name}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    width: "100%",
    justifyContent: "center",
    alignItems: "center",
    borderTopColor: theme.colors.batteryChargedBlue,
    paddingTop: 10  },
  labelStyle: {
    fontSize: 16,
    fontStyle:'normal',
    fontWeight:'bold',
  },
  activeColor:{
    color:theme.colors.bondiBlue,
  },
  inactiveColor:{
    color:theme.colors.zinnwalditeBrown,
  }
});
