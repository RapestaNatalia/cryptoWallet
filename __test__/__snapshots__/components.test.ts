import React from "react";
import { Logo } from "../../components/ui/Logo";
import { Fetching } from "../../components/ui/ActivityIndicator";
import { TabNavBar } from "../../components/ui/TabNavBar";
import renderer from "react-test-renderer";
import { PrimaryButton } from "../../components/ui/PrimaryButton";
import { HeaderLeft } from "../../components/ui/HeaderLeft";
import { CoinInfo } from "../../components/ui/CoinInfo";
describe("Snapshot tests", () => {
  it("Logo snapShot", () => {
    const tree = renderer.create(Logo()).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Activity Indicator snapShot", () => {
    const tree = renderer
      .create(Fetching({ animating: true, color: "blue" }))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("TabNavBar snapShot", () => {
    const tree = renderer
      .create(TabNavBar({ isFocus: true, name: "To Do" }))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("PrimaryButton snapShot", () => {
    const tree = renderer
      .create(
        PrimaryButton({
          title: "Get Started",
          onPress: () => {},
          accessibilityHint: "Press this button to start session",
          accessibilityLabel: "This is the LogIn button",
        })
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("HeaderLeft snapShot", () => {
    const tree = renderer
      .create(HeaderLeft({ onPress: () => {}, name: "" }))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("CoinInfo snapShot", () => {
    const item = {
      changePercent24Hr: "-4.8350931871279116",
      explorer: "https://explorer.binance.org/asset/RUNE-B1A",
      id: "thorchain",
      marketCapUsd: "1935142965.0384600300689446",
      maxSupply: "500000000.0000000000000000",
      name: "THORChain",
      priceUsd: "5.8518682447421885",
      rank: "65",
      supply: "330688061.3344559700000000",
      symbol: "RUNE",
      volumeUsd24Hr: "47598877.9465108057755086",
      vwap24Hr: "5.9481127113701395",
    };
    const tree = renderer
      .create(CoinInfo({ onPress: () => {}, item }))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
