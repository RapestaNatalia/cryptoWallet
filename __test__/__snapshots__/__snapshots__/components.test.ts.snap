// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests Activity Indicator snapShot 1`] = `
<ActivityIndicator
  animating={true}
  color="blue"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`Snapshot tests CoinInfo snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderRadius": 8,
        "display": "flex",
        "marginVertical": 6,
        "padding": 8,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#0A132C",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "bold",
              }
            }
          >
            RUNE
             - 
          </Text>
          <Text
            style={
              Object {
                "color": "#0A132C",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "bold",
              }
            }
          >
            THORChain
          </Text>
        </View>
        <View
          style={Object {}}
        >
          <Text
            style={
              Object {
                "alignSelf": "flex-end",
                "color": "#6B7280",
                "fontSize": 14,
                "fontStyle": "normal",
              }
            }
          >
            #
            65
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#019FB5",
                "fontSize": 24,
                "fontWeight": "normal",
              }
            }
          >
            $ 
            5.85
            <Text
              style={
                Object {
                  "color": "#6B7280",
                  "fontSize": 14,
                  "textAlignVertical": "bottom",
                }
              }
            >
               USD
            </Text>
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "flex-end",
              },
              Object {
                "alignItems": "flex-end",
              },
              Object {
                "backgroundColor": "#FDDCDC",
                "color": "#A50606",
              },
              Object {
                "alignItems": "center",
                "borderRadius": 12,
                "fontSize": 14,
                "fontStyle": "normal",
                "fontWeight": "500",
                "paddingHorizontal": 10,
                "paddingVertical": 2,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "transform": Array [
                  Object {
                    "rotate": "180deg",
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              bbHeight={12}
              bbWidth={10}
              fill="none"
              focusable={false}
              height={12}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 12,
                    "width": 10,
                  },
                ]
              }
              width={10}
            >
              <RNSVGGroup
                fill={null}
                propList={
                  Array [
                    "fill",
                  ]
                }
              >
                <RNSVGPath
                  clipRule={0}
                  d="M.293 5.707a1 1 0 0 1 0-1.414l4.005-4a1.002 1.002 0 0 1 1.416 0l4.005 4a1 1 0 0 1-1.416 1.414L6.007 3.414V11a1 1 0 0 1-1.001 1 1.002 1.002 0 0 1-1.001-1V3.414L1.709 5.707a1.002 1.002 0 0 1-1.416 0Z"
                  fill={4294130234}
                  fillRule={0}
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <Text
            style={
              Object {
                "alignItems": "flex-end",
                "justifyContent": "flex-end",
              }
            }
          >
             
            4.8
            %
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Snapshot tests HeaderLeft snapShot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "paddingLeft": 10,
      }
    }
  >
    <RNSVGSvgView
      bbHeight={10}
      bbWidth={6}
      fill="none"
      focusable={false}
      height={10}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "flex": 0,
            "height": 10,
            "width": 6,
          },
        ]
      }
      width={6}
    >
      <RNSVGGroup
        fill={null}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          clipRule={0}
          d="M5.707.293a1 1 0 0 1 0 1.414L2.414 5l3.293 3.293a1 1 0 0 1-1.414 1.414l-4-4a1 1 0 0 1 0-1.414l4-4a1 1 0 0 1 1.414 0Z"
          fill={4278296501}
          fillRule={0}
          propList={
            Array [
              "fill",
              "fillRule",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <Text
      style={
        Object {
          "color": "#019FB5",
          "fontSize": 14,
          "marginLeft": 7,
        }
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Snapshot tests Logo snapShot 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../assets/logo.png",
    }
  }
  style={
    Object {
      "height": 26.07,
      "width": 28.53,
    }
  }
/>
`;

exports[`Snapshot tests PrimaryButton snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#1FC4DB",
      "borderRadius": 6,
      "height": 42,
      "marginTop": 44,
      "width": "100%",
    }
  }
>
  <View
    accessibilityHint="Press this button to start session"
    accessibilityLabel="This is the LogIn button"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "padding": 9,
          "textAlign": "center",
        }
      }
    >
      Get Started
    </Text>
  </View>
</View>
`;

exports[`Snapshot tests TabNavBar snapShot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderTopColor": "#1FC4DB",
        "justifyContent": "center",
        "paddingTop": 10,
        "width": "100%",
      },
      Object {
        "borderTopWidth": 3,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "bold",
        },
        Object {
          "color": "#019FB5",
        },
      ]
    }
  >
    To Do
  </Text>
</View>
`;
